File: updated_code/code/python/1556-thousand-separator.py
"""
This is a solution to add a dot (.) as the thousands separator and return it in string format.

Example 1:
    Input: n = 987    Output: "987"

Example 2:
    Input: n = 1234    Output: "1.234"

Constraints:
    0 <= n <= 231 - 1
"""
class Solution:
    """
    This function takes an integer `n` as input, converts it to a string, and adds a dot (.) as the thousands separator.
    It then returns the result as a string.

    :param n: The integer to add a dot (.) as the thousands separator.
    :return: The integer with a dot (.) as the thousands separator as a string.
    """
    def thousand_separator(self, n: int) -> str:
        n_str = str(n)
        result = ""
        for i, char in enumerate(reversed(n_str)):
            if i % 3 == 0 and i != 0:
                result = "." + result
            result = char + result
        return "0" + result if n != 0 else "0"

# add a newline at the end

File: updated_code/code/java/1556-thousand-separator.java
package com.problem;

/**
 * Given an integer `n`, add a dot (".") as the thousands separator and return it in string format.
 */
public class Solution {
    /**
     * Add a dot (".") as the thousands separator and return it in string format.
     * 
     * @param n the input integer
     * @return the string with thousands separator
     */
    public String thousandSeparator(int n) {
        final String THOUSAND_SEPARATOR = ".";
        if (n < 1000) {
            return String.valueOf(n);
        }
        String str = String.valueOf(n);
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < str.length(); i++) {
            sb.append(str.charAt(i));
            if ((i + 1) % 3 == 0 && i != str.length() - 1) {
                sb.append(THOUSAND_SEPARATOR);
            }
        }
        return sb.toString();
    }
}